{"id": "ROOT", "speaker": "Devligue", "conversation_id": "ROOT", "reply_to": "ROOT", "timestamp": "2018-06-03T18:25:47Z", "text": "Unable to delete files on Fedora 28 (gvfs-trash is deprecated) <!--\r \r Have you read Atom's Code of Conduct? By filing an Issue, you are expected to comply with it, including treating everyone with respect: https://github.com/atom/atom/blob/master/CODE_OF_CONDUCT.md\r \r Do you want to ask a question? Are you looking for support? The Atom message board is the best place for getting support: https://discuss.atom.io\r \r -->\r \r ### Prerequisites\r \r * [x] Put an X between the brackets on this line if you have done all of the following:\r     * Reproduced the problem in Safe Mode: https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode\r     * Followed all applicable steps in the debugging guide: https://flight-manual.atom.io/hacking-atom/sections/debugging/\r     * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq\r     * Checked that your issue isn't already filed: https://github.com/issues?utf8=\u2713&q=is%3Aissue+user%3Aatom\r     * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages\r \r ### Description\r \r My issue is the same as in #15949 except it is not fixed. \r \r I am unable to delete file (or directory). When trying, the following message appears: `Is gvfs-trash installed?`\r \r ### Steps to Reproduce\r \r 1. Try to delete any file or directory\r 2. ???\r 3. Profit\r \r **Expected behavior:** Deleting file should delete file\r \r **Actual behavior:** The file is not deleted\r \r **Reproduces how often:** Every time\r \r ### Versions\r \r ```\r Atom    : 1.27.1\r Electron: 1.7.15\r Chrome  : 58.0.3029.110\r Node    : 7.9.0\r ```\r ```\r apm  1.19.0\r npm  3.10.10\r node 6.9.5 x64\r atom 1.27.1\r python 2.7.15\r git 2.17.1\r ```\r ```\r OS: Fedora 28\r ```\r \r ### Additional Information\r ```\r $ gvfs-trash\r This tool has been deprecated, use 'gio trash' instead.\r See 'gio help trash' for more info.\r ```\r The `gio trash` is supposedly implemented since electron 1.7.2 and for some folks out there this problem was fixed with `Atom 1.25` (which included upgrade to `Electron 1.7.11`) but apparently I am on even newer Atom version, with even newer Electron and it still happens.\r \r ### EDIT1\r #### More insight on the problem provided by @cutephoton:\r > @Devligue This is a legit bug. The _tldr answer_ is that `_g_file_trash` may not be supported. In order for `_g_file_trash` to succeed, the following conditions must be met:\r > \r > * The file being moved to trash is on the same partition as your home folder\r >   **OR**\r > * A trash folder already exists or can be created at the mount point and,\r > * The trash folder is considered 'safe/secure' by verifying UID and restrictive permissions\r > \r > **How this should probably be fixed...**\r > \r > Atom does not have a fallback mechanism like offering the ability to permanently remove a file instead. This would address bring atom in line with gnome's graceful behaviour. The attached image show how Gnome Files (aka Nautilus) prompt the user to permanently delete the file (when trash not supported)\r > \r > ![gnome's solution](https://user-images.githubusercontent.com/19512121/43511743-0dac32c0-952e-11e8-8af8-e0ef6cb916d0.png)\r > \r > **Impact: Users with multiple disks and partitions**\r > \r > Fedora/Redhat/etc: _Impacted_ due to default partition scheme that separates `/` and `/home` in to partitions. Files outside of the `/home` partition cannot be moved to trash.\r > \r > Ubuntu: Less likely due to partition layout (`/home` is part of the `/` partition)\r > \r > Conditions where users will be impacted:\r > \r > * File systems with unix permissions:\r >   \r >   * _Impacted with workaround_ due to typical restrictive top-level directory permissions (root)\r > * FAT/non-unix file systems:\r >   \r >   * User Session Mount (i.e. `/run/media/$USER/disk-label`): _No Impact_\r >   * Fstab: _Impacted_ even with permissive umask. The default uid/gid is root. It will be unable to satisfy the requirements of a trash folder without additional options (uid/gid/umask).\r > * Network shared folders: _Unknown/Did not test._\r > \r > **Notes/Testing the Root Cause**\r > \r > Note: Though I am confident that my analysis is decent enough, much of the code I was referencing was unfamiliar/new to me.\r > \r > Initially I encountered this issue when I put files in a certain location like the reporter above. I created a delete-me file test as suggested above in the relevant directory, `/opt/cupenv`. I ran the command `strace gio trash delete-me.txt`. A abridged version of the output is here:\r > \r > ```\r > lstat(\"/opt/cupenv/delete-me.txt\", {st_mode=S_IFREG|0664, st_size=6, ...}) = 0\r > ...\r > lstat(\"/.Trash-1000\", 0x7ffde5ffb5b0)   = -1 ENOENT (No such file or directory)\r > mkdir(\"/.Trash-1000\", 0700)             = -1 EACCES (Permission denied)\r > ```\r > That's unexpected. The API documentation for [g_file_trash](https://developer.gnome.org/gio/stable/GFile.html#g-file-trash) lacks some level of specificity.\r > \r > > Sends file to the \"Trashcan\", if possible. This is similar to deleting it, but the user can recover it before emptying the trashcan. Not all file systems support trashing, so this call can return the G_IO_ERROR_NOT_SUPPORTED error.\r > \r > One might assume (as I did): Given a path, if the user has permission to modify/delete the file, then GIO's g_file_trash API should succeed at removing the file in some manner. Perhaps, if trash functionality is not available, then there might be a fallback mechanism. In the case of `gio trash` I expected there to be a -f force option that would prioritize trash over permanently deleting the file. (`gio trash -f` only ignores files that don't exist)\r > \r > Glib appears to implement GIO local file access using [glocalfile.c](https://github.com/GNOME/glib/blob/83a4cab12c2d00dbfe6013d071cff2da310109a4/gio/glocalfile.c#L1899). The trash algorithm looks like this:\r > \r > * Is the path on the same partition as the user's home directory? If so, move the file to the home trash folder if possible and exit.\r > * Given the path, locate the mount point top level directory (denoted as `$topdir` in source)\r >   \r >   * In my case, `/opt/cupenv/delete_me.txt` -> `/` (the primary partition)\r >   * A more typical case: `/run/media/$USER/disk-label/a/b/c/d` would be `/run/media/$USER/disk-label`\r > * If `$topdir` is found, pick one of `$topdir/.Trash/$UID` or `$topdir/.Trash-UID` folder.\r >   \r >   * If `$topdir/.Trash-UID` and `$topdir/.Trash/$UID` exists: Validate proper UID and file permission (or fail)\r >   * Try to create folder `$topdir/.Trash-$UID`. Validate proper UID and file permission (or fail)\r > \r > At the end of the function we find [`G_IO_ERROR_NOT_SUPPORTED` is returned](https://github.com/GNOME/glib/blob/83a4cab12c2d00dbfe6013d071cff2da310109a4/gio/glocalfile.c#L2059) when such a folder is not found and cannot be created.\r > \r > ```\r > if (trashdir == NULL)\r > \t{\r > \t  g_free (topdir);\r >           g_set_io_error (error,\r >                           _(\"Unable to find or create trash directory for %s\"),\r >                           file, G_IO_ERROR_NOT_SUPPORTED);\r > \t  return FALSE;\r > }\r > ```\r > I confirmed my understanding by creating a top level trash folder and using gio trash command.\r > \r > ```\r > sudo mkdir /.Trash-1000\r > sudo chmod 0700 /.Trash-1000\r > sudo chown fosterb:fosterb /.Trash-1000\r > cd /opt/cupenv && echo hello > delete-me.txt; gio trash delete-me.txt\r > ```\r > Files located on the same partition as your home directory can always be moved to your user trash folder. On my operating system, the paths `/` and `/home` are separate partitions. I created a file `/home/tmp/x` where `tmp` and `x` are owned by my user. In this case, the files get moved to your `$HOME/.local/share/Trash` folder.\r > \r > ```\r > lstat(\"/home/tmp/x\", {st_mode=S_IFREG|0644, st_size=0, ...}) = 0\r > stat(\"/home/fosterb\", {st_mode=S_IFDIR|0700, st_size=4096, ...}) = 0\r > access(\"/home\", F_OK)                   = 0\r > ....\r > stat(\"/home/fosterb/.local/share\", {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0\r > access(\"/home/fosterb/.local/share/Trash\", F_OK) = 0\r > stat(\"/home/fosterb/.local/share/Trash\", {st_mode=S_IFDIR|0700, st_size=4096, ...}) = 0\r > mkdir(\"/home/fosterb/.local/share/Trash/info\", 0700) = -1 EEXIST (File exists)\r > mkdir(\"/home/fosterb/.local/share/Trash/files\", 0700) = -1 EEXIST (File exists)\r > openat(AT_FDCWD, \"/home/fosterb/.local/share/Trash/info/x.trashinfo\", O_RDONLY|O_CREAT|O_EXCL, 0666) = 7\r > ```\r > But when mounting disks (external or internal) using fstab or mount commands (as opposed to session-based mounting) issues can arise. I tested a FAT file system with umask=0000 and uid/gid set to root.\r > \r > ```\r > lstat(\"/mnt/giotest\", {st_mode=S_IFDIR|0777, st_size=8192, ...}) = 0\r > lstat(\"/mnt\", {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0\r > lstat(\"/mnt/giotest/.Trash\", 0x7fffec5502c0) = -1 ENOENT (No such file or directory)\r > lstat(\"/mnt/giotest/.Trash-1000\", 0x7fffec550230) = -1 ENOENT (No such file or directory)\r > mkdir(\"/mnt/giotest/.Trash-1000\", 0700) = 0\r > lstat(\"/mnt/giotest/.Trash-1000\", {st_mode=S_IFDIR|0777, st_size=8192, ...}) = 0\r > unlink(\"/mnt/giotest/.Trash-1000\")      = -1 EISDIR (Is a directory)\r > rmdir(\"/mnt/giotest/.Trash-1000\")       = 0\r > file:///mnt/giotest/x: Unable to find or create trash directory for /mnt/giotest/x\r > ```\r > You can see gio trash going through the motions. The operations succeeds but the command still reports an error.\r > \r > ```\r > [fosterb@rose giotest]$ ls -lah\r > total 12K\r > drwxrwxrwx. 2 root root 8.0K Aug  1 00:36 .\r > drwxr-xr-x. 4 root root 4.0K Aug  1 00:30 ..\r > -rwxrwxrwx. 1 root root    0 Aug  1 01:23 x\r > ```\r > GLib/Gio will not write to a trash directory with incorrect permissions/ownership. (Security?) Without unix permissions, trash will always fail on these mount points. Modifying fstab to appear similar to options used during session mounts (UID/GID set to user vs root, umask is set appropriately). Conveniently the [source code](https://github.com/GNOME/glib/blob/83a4cab12c2d00dbfe6013d071cff2da310109a4/gio/glocalfile.c#L2046) has a comment that seems to strongly imply this is a known/expected.\r > \r > ```\r > \t\t  /* Ensure that the created dir has the right uid etc.\r > \t\t     This might fail on e.g. a FAT dir */\r > ```\r > Most of this was unnecessary, but I figured I'd show my work.\r \r ### EDIT2\r #### More insight, to not get confused as of the nature of the bug, and how to reproduce it (by @cutephoton as well):\r > Yes, as reported, the ELECTRON_TRASH environment variable is needed. This is the issue referenced #15949 bug. The issue reported here is not related to #15949. The error message related to gvfs-trash is a blunt tool (even access denied errors produce the same error message) so I understand why people are responding here.\r > \r > The reporter clarifies one of the key symptoms here:\r > [#17452 (comment)](https://github.com/atom/atom/issues/17452#issuecomment-396407349)\r > \r > And that led to my deep dive. My deep dive details the specific cases where trash becomes unavailable (i.e. when a trash folder cannot be safely created).\r > [#17452 (comment)](https://github.com/atom/atom/issues/17452#issuecomment-409518547)\r > \r > **I just revalidated my findings and unsurprisingly gio trash still fails.**\r > \r > Anyhow, I wanted to clarify this so it doesn't get lost when/if atom devs look in to this issue.\r \r \r ", "meta": {"posReactions": "5", "negReactions": "0"}}
{"id": "COM0", "speaker": "rsese", "conversation_id": "ROOT", "reply_to": "ROOT", "timestamp": "2018-06-04T20:01:19Z", "text": "Thanks for opening a new issue - is `gio` available on Fedora 28?  ", "meta": {"posReactions": "0", "negReactions": "0"}}
{"id": "COM1", "speaker": "Devligue", "conversation_id": "ROOT", "reply_to": "COM0", "timestamp": "2018-06-04T20:06:35Z", "text": "Yes, `gio` is available and installed by default", "meta": {"posReactions": "0", "negReactions": "0"}}
{"id": "COM2", "speaker": "rsese", "conversation_id": "ROOT", "reply_to": "COM1", "timestamp": "2018-06-05T02:12:33Z", "text": "Hmmm, not sure why it's not working then - any difference if you specifically set an `ELECTRON_TRASH` environment variable to `gio` and then restart Atom?", "meta": {"posReactions": "0", "negReactions": "0"}}
{"id": "COM3", "speaker": "Devligue", "conversation_id": "ROOT", "reply_to": "COM2", "timestamp": "2018-06-05T22:31:29Z", "text": "It changes nothing :thinking: \r\n\r\n```\r\n$ gio --version\r\n2.56.1\r\n```\r\n\r\n```\r\necho $ELECTRON_TRASH\r\ngio\r\n```\r\n", "meta": {"posReactions": "0", "negReactions": "0"}}
{"id": "COM4", "speaker": "rsese", "conversation_id": "ROOT", "reply_to": "COM3", "timestamp": "2018-06-11T21:35:37Z", "text": "And if you just use `gio trash` from the command line it works fine?  E.g.:\r\n\r\n```\r\necho hello > delete-me.txt\r\ngio trash delete-me.txt\r\n```\r\n\r\nAlso, are you getting the exact same error message you mentioned (\"When trying, the following message appears: Is gvfs-trash installed?\") after you set the `ELECTRON_TRASH` environment variable to `gio`?  ", "meta": {"posReactions": "1", "negReactions": "0"}}
{"id": "COM5", "speaker": "Devligue", "conversation_id": "ROOT", "reply_to": "COM4", "timestamp": "2018-06-11T22:28:55Z", "text": "The `echo hello > delete-me.txt; gio trash delete-me.txt` works just fine normally but crashes on the ntfs partition. As I see this, it is not an Atom bug, however the error message it kept returning me was quite misleading.\r\n\r\nI didn't solve the problem yet, but I am closing this since it is not related to Atom. Sorry for trouble!", "meta": {"posReactions": "1", "negReactions": "0"}}
{"id": "COM6", "speaker": "50Wliu", "conversation_id": "ROOT", "reply_to": "COM5", "timestamp": "2018-06-12T01:13:03Z", "text": "> however the error message it kept returning me was quite misleading.\r\n\r\nYeah, that's the default error message.  Unfortunately, Electron's moveToTrash method doesn't say _why_ it failed, just that it did.  We've found that the most common reason is that `gvfs-trash` isn't installed, which is why we decided to include it in the error message.", "meta": {"posReactions": "0", "negReactions": "0"}}
{"id": "COM7", "speaker": "cutephoton", "conversation_id": "ROOT", "reply_to": "COM6", "timestamp": "2018-08-01T09:48:16Z", "text": "@Devligue This is a legit bug. The _tldr answer_ is that `_g_file_trash` may not be supported. In order for `_g_file_trash` to succeed, the following conditions must be met:\r\n\r\n- The file being moved to trash is on the same partition as your home folder\r\n**OR**\r\n- A trash folder already exists or can be created at the mount point and,\r\n- The trash folder is considered 'safe/secure' by verifying UID and restrictive permissions\r\n\r\n**How this should probably be fixed...**\r\n\r\nAtom does not have a fallback mechanism like offering the ability to permanently remove a file instead. This would address bring atom in line with gnome's graceful behaviour. The attached image show how  Gnome Files (aka Nautilus) prompt the user to permanently delete the file (when trash not supported)\r\n\r\n![gnome's solution](https://user-images.githubusercontent.com/19512121/43511743-0dac32c0-952e-11e8-8af8-e0ef6cb916d0.png)\r\n\r\n**Impact: Users with multiple disks and partitions**\r\n\r\nFedora/Redhat/etc: _Impacted_ due to default partition scheme that separates `/` and `/home` in to partitions. Files outside of the `/home` partition cannot be moved to trash.\r\n\r\nUbuntu: Less likely due to partition layout (`/home` is part of the `/` partition)\r\n\r\nConditions where users will be impacted:\r\n- File systems with unix permissions:\r\n  - _Impacted with workaround_ due to typical restrictive top-level directory permissions (root)\r\n- FAT/non-unix file systems:\r\n  - User Session Mount (i.e. `/run/media/$USER/disk-label`): _No Impact_\r\n  - Fstab: _Impacted_ even with permissive umask. The default uid/gid is root. It will be unable to satisfy the requirements of a trash folder without additional options (uid/gid/umask).\r\n- Network shared folders: _Unknown/Did not test._\r\n\r\n**Notes/Testing the Root Cause**\r\n\r\nNote: Though I am confident that my analysis is decent enough, much of the code I was referencing was unfamiliar/new to me.\r\n\r\nInitially I encountered this issue when I put files in a certain location like the reporter above. I created a delete-me file test as suggested above in the relevant directory, `/opt/cupenv`. I ran the command  `strace gio trash delete-me.txt`. A abridged version of the output is here:\r\n\r\n```\r\nlstat(\"/opt/cupenv/delete-me.txt\", {st_mode=S_IFREG|0664, st_size=6, ...}) = 0\r\n...\r\nlstat(\"/.Trash-1000\", 0x7ffde5ffb5b0)   = -1 ENOENT (No such file or directory)\r\nmkdir(\"/.Trash-1000\", 0700)             = -1 EACCES (Permission denied)\r\n```\r\nThat's unexpected. The API documentation for [g_file_trash](https://developer.gnome.org/gio/stable/GFile.html#g-file-trash) lacks some level of specificity.\r\n\r\n> Sends file to the \"Trashcan\", if possible. This is similar to deleting it, but the user can recover it before emptying the trashcan. Not all file systems support trashing, so this call can return the G_IO_ERROR_NOT_SUPPORTED error. \r\n\r\nOne might assume (as I did): Given a path, if the user has permission to modify/delete the file, then GIO's g_file_trash API should succeed at removing the file in some manner. Perhaps, if trash functionality is not available, then there might be a fallback mechanism. In the case of `gio trash` I expected there to be a -f force option that would prioritize trash over permanently deleting the file. (`gio trash -f` only ignores files that don't exist)\r\n\r\nGlib appears to implement GIO local file access using [glocalfile.c](https://github.com/GNOME/glib/blob/83a4cab12c2d00dbfe6013d071cff2da310109a4/gio/glocalfile.c#L1899). The trash algorithm looks like this:\r\n\r\n- Is the path on the same partition as the user's home directory? If so, move the file to the home trash folder if possible and exit.\r\n- Given the path, locate the mount point top level directory (denoted as `$topdir` in source)\r\n  - In my case, `/opt/cupenv/delete_me.txt` -> `/` (the primary partition)\r\n  - A more typical case: `/run/media/$USER/disk-label/a/b/c/d` would be `/run/media/$USER/disk-label`\r\n- If `$topdir` is found, pick one of `$topdir/.Trash/$UID` or `$topdir/.Trash-UID` folder.\r\n  - If `$topdir/.Trash-UID` and `$topdir/.Trash/$UID` exists: Validate proper UID and file permission (or fail)\r\n  - Try to create folder `$topdir/.Trash-$UID`. Validate proper UID and file permission (or fail)\r\n\r\nAt the end of the function we find [`G_IO_ERROR_NOT_SUPPORTED` is returned](https://github.com/GNOME/glib/blob/83a4cab12c2d00dbfe6013d071cff2da310109a4/gio/glocalfile.c#L2059) when such a folder is not found and cannot be created.\r\n\r\n```\r\nif (trashdir == NULL)\r\n\t{\r\n\t  g_free (topdir);\r\n          g_set_io_error (error,\r\n                          _(\"Unable to find or create trash directory for %s\"),\r\n                          file, G_IO_ERROR_NOT_SUPPORTED);\r\n\t  return FALSE;\r\n}\r\n```\r\n\r\nI confirmed my understanding by creating a top level trash folder and using gio trash command.\r\n\r\n```\r\nsudo mkdir /.Trash-1000\r\nsudo chmod 0700 /.Trash-1000\r\nsudo chown fosterb:fosterb /.Trash-1000\r\ncd /opt/cupenv && echo hello > delete-me.txt; gio trash delete-me.txt\r\n```\r\nFiles located on the same partition as your home directory can always be moved to your user trash folder. On my operating system, the paths `/` and `/home` are separate partitions. I created a file `/home/tmp/x` where `tmp` and `x` are owned by my user. In this case, the files get moved to your `$HOME/.local/share/Trash` folder.\r\n\r\n```\r\nlstat(\"/home/tmp/x\", {st_mode=S_IFREG|0644, st_size=0, ...}) = 0\r\nstat(\"/home/fosterb\", {st_mode=S_IFDIR|0700, st_size=4096, ...}) = 0\r\naccess(\"/home\", F_OK)                   = 0\r\n....\r\nstat(\"/home/fosterb/.local/share\", {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0\r\naccess(\"/home/fosterb/.local/share/Trash\", F_OK) = 0\r\nstat(\"/home/fosterb/.local/share/Trash\", {st_mode=S_IFDIR|0700, st_size=4096, ...}) = 0\r\nmkdir(\"/home/fosterb/.local/share/Trash/info\", 0700) = -1 EEXIST (File exists)\r\nmkdir(\"/home/fosterb/.local/share/Trash/files\", 0700) = -1 EEXIST (File exists)\r\nopenat(AT_FDCWD, \"/home/fosterb/.local/share/Trash/info/x.trashinfo\", O_RDONLY|O_CREAT|O_EXCL, 0666) = 7\r\n```\r\nBut when mounting disks (external or internal) using fstab or mount commands (as opposed to session-based mounting) issues can arise. I tested a FAT file system with umask=0000 and uid/gid set to root.\r\n\r\n```\r\nlstat(\"/mnt/giotest\", {st_mode=S_IFDIR|0777, st_size=8192, ...}) = 0\r\nlstat(\"/mnt\", {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0\r\nlstat(\"/mnt/giotest/.Trash\", 0x7fffec5502c0) = -1 ENOENT (No such file or directory)\r\nlstat(\"/mnt/giotest/.Trash-1000\", 0x7fffec550230) = -1 ENOENT (No such file or directory)\r\nmkdir(\"/mnt/giotest/.Trash-1000\", 0700) = 0\r\nlstat(\"/mnt/giotest/.Trash-1000\", {st_mode=S_IFDIR|0777, st_size=8192, ...}) = 0\r\nunlink(\"/mnt/giotest/.Trash-1000\")      = -1 EISDIR (Is a directory)\r\nrmdir(\"/mnt/giotest/.Trash-1000\")       = 0\r\nfile:///mnt/giotest/x: Unable to find or create trash directory for /mnt/giotest/x\r\n```\r\n\r\nYou can see gio trash going through the motions. The operations succeeds but the command still reports an error.\r\n```\r\n[fosterb@rose giotest]$ ls -lah\r\ntotal 12K\r\ndrwxrwxrwx. 2 root root 8.0K Aug  1 00:36 .\r\ndrwxr-xr-x. 4 root root 4.0K Aug  1 00:30 ..\r\n-rwxrwxrwx. 1 root root    0 Aug  1 01:23 x\r\n```\r\n\r\nGLib/Gio will not write to a trash directory with incorrect permissions/ownership. (Security?) Without unix permissions, trash will always fail on these mount points. Modifying fstab to appear similar to options used during session mounts (UID/GID set to user vs root, umask is set appropriately). Conveniently the [source code](https://github.com/GNOME/glib/blob/83a4cab12c2d00dbfe6013d071cff2da310109a4/gio/glocalfile.c#L2046) has a comment that seems to strongly imply this is a known/expected.\r\n\r\n```\r\n\t\t  /* Ensure that the created dir has the right uid etc.\r\n\t\t     This might fail on e.g. a FAT dir */\r\n```\r\nMost of this was unnecessary, but I figured I'd show my work.", "meta": {"posReactions": "2", "negReactions": "0"}}
{"id": "COM8", "speaker": "yodatak", "conversation_id": "ROOT", "reply_to": "COM7", "timestamp": "2018-10-31T22:49:15Z", "text": "I got the same problem on feodra 29 on a fresh install can we reopen this bug ?", "meta": {"posReactions": "13", "negReactions": "0"}}
{"id": "COM9", "speaker": "cutephoton", "conversation_id": "ROOT", "reply_to": "COM8", "timestamp": "2018-10-31T23:14:47Z", "text": "@Devligue can reopen this", "meta": {"posReactions": "0", "negReactions": "0"}}
{"id": "COM10", "speaker": "no-response[bot]", "conversation_id": "ROOT", "reply_to": "COM9", "timestamp": "2018-11-01T14:20:10Z", "text": "This issue has been automatically closed because there has been no response to our request for more information from the original author. With only the information that is currently in the issue, we don't have enough information to take action. Please reach out if you have or find the answers we need so that we can investigate further.\n", "meta": {"posReactions": "0", "negReactions": "0"}}
{"id": "COM11", "speaker": "Devligue", "conversation_id": "ROOT", "reply_to": "COM10", "timestamp": "2018-11-01T14:52:10Z", "text": "I may have no additional information about the issue, but I am sure @cutephoton provided enough insight. There are also other people who have similar unsolved problems. I am reopening this.", "meta": {"posReactions": "10", "negReactions": "0"}}
{"id": "COM12", "speaker": "yodatak", "conversation_id": "ROOT", "reply_to": "COM11", "timestamp": "2018-11-05T22:24:01Z", "text": "I got no problem with the flatpak version by the way", "meta": {"posReactions": "0", "negReactions": "0"}}
{"id": "COM13", "speaker": "cutephoton", "conversation_id": "ROOT", "reply_to": "COM12", "timestamp": "2018-11-06T00:54:29Z", "text": "I reproduced on the latest version still. **shrug** I don't see how flatpak and rpm/deb would differ. It only fails when the preconditions listed above are met -- a partition where you don't have permission to create a trash folder under your user ID.", "meta": {"posReactions": "0", "negReactions": "0"}}
{"id": "COM14", "speaker": "yodatak", "conversation_id": "ROOT", "reply_to": "COM13", "timestamp": "2018-11-06T21:31:27Z", "text": "With flatpak i got no error but it don't get delete sorry about this misinformation. I try to disable selinux it don't change.", "meta": {"posReactions": "0", "negReactions": "0"}}
{"id": "COM15", "speaker": "cutephoton", "conversation_id": "ROOT", "reply_to": "COM14", "timestamp": "2018-11-06T21:48:03Z", "text": "No. Not selinux. It has to do with how trash folders work. My rather verbose explanation gets in to the nitty gritty for where atom's behavior is deficient.  The short answer is that atom only asks to move files to trash and does not delete files. If a trash folder cannot be created (multiple causes all related to permissions) than it fails. Atom should inquire with user if they should permanently delete the files instead of failing.  ", "meta": {"posReactions": "0", "negReactions": "0"}}
{"id": "COM16", "speaker": "mkungla", "conversation_id": "ROOT", "reply_to": "COM15", "timestamp": "2018-11-13T11:29:03Z", "text": "As @yodatak also mentioned that Fedora 29 fresh install has this problem since `gvfs-trash` has finally after many deprecation notices been removed  and replaced with  `gio`. So as @rsese mentioned you need to set `ELECTRON_TRASH=gio` e.g. add it to your `~/.bash_profile`. \r\n\r\nlogout and login or `source ~/.bash_profile` and trashing will work\r\n\r\nor\r\n```\r\necho 'export ELECTRON_TRASH=gio' >> $HOME/.bashrc\r\nsource $HOME/.bashrc\r\n```", "meta": {"posReactions": "14", "negReactions": "1"}}
{"id": "COM17", "speaker": "cutephoton", "conversation_id": "ROOT", "reply_to": "COM16", "timestamp": "2018-11-13T12:54:30Z", "text": "Gio operates the same way. I used the Gio command line to manually step through this...", "meta": {"posReactions": "0", "negReactions": "0"}}
{"id": "COM18", "speaker": "mh-cbon", "conversation_id": "ROOT", "reply_to": "COM17", "timestamp": "2018-12-06T12:06:34Z", "text": "running fedora 29, i have installed atom from packages, since recently I faced the same error as described here.\r\n\r\n```\r\nsudo dnf list installed atom\r\nPaquets install\u00e9s\r\natom.x86_64                                                1.33.0-0.1                                                 @Atom\r\n```\r\n\r\nI think this bug deserves an action in order to improve the package installation, or the runtime, to detect gio/gvfs and make this work out of the box.\r\n\r\nThis really is an ugly bug for the end user experience.", "meta": {"posReactions": "8", "negReactions": "0"}}
{"id": "COM19", "speaker": "cristianlivella", "conversation_id": "ROOT", "reply_to": "COM18", "timestamp": "2018-12-09T12:32:04Z", "text": "Is there any workaround to delete files with Atom on Fedora 29?", "meta": {"posReactions": "1", "negReactions": "0"}}
{"id": "COM20", "speaker": "edlefebvre", "conversation_id": "ROOT", "reply_to": "COM19", "timestamp": "2018-12-12T08:32:45Z", "text": "> So as @rsese mentioned you need to set `ELECTRON_TRASH=gio` e.g. add it to your `~/.bash_profile`.\r\n> \r\n> logout and login or `source ~/.bash_profile` and trashing will work\r\n\r\nThis is not working on fedora 29, even with gio installed and ELECTRON_TRASH variable set to gio.", "meta": {"posReactions": "1", "negReactions": "0"}}
{"id": "COM21", "speaker": "cutephoton", "conversation_id": "ROOT", "reply_to": "COM20", "timestamp": "2018-12-12T14:03:28Z", "text": "There are no workarounds.", "meta": {"posReactions": "0", "negReactions": "0"}}
{"id": "COM22", "speaker": "olegjs", "conversation_id": "ROOT", "reply_to": "COM21", "timestamp": "2018-12-17T19:44:08Z", "text": "> > So as @rsese mentioned you need to set `ELECTRON_TRASH=gio` e.g. add it to your `~/.bash_profile`.\r\n> > logout and login or `source ~/.bash_profile` and trashing will work\r\n> \r\n> This is not working on fedora 29, even with gio installed and ELECTRON_TRASH variable set to gio.\r\n\r\nWorks on my Fedora 29 -- had to restart terminal and restart atom from that terminal.", "meta": {"posReactions": "2", "negReactions": "0"}}
{"id": "COM23", "speaker": "cutephoton", "conversation_id": "ROOT", "reply_to": "COM22", "timestamp": "2018-12-17T20:30:24Z", "text": "That is a not a fix. There are preconditions as documented above to reproduce the bug.", "meta": {"posReactions": "0", "negReactions": "0"}}
{"id": "COM24", "speaker": "shivdhar", "conversation_id": "ROOT", "reply_to": "COM23", "timestamp": "2018-12-28T08:30:30Z", "text": "Would love for this to be fixed (on my Fedora 29 installed via the official RPM).", "meta": {"posReactions": "1", "negReactions": "0"}}
{"id": "COM25", "speaker": "histamineblkr", "conversation_id": "ROOT", "reply_to": "COM24", "timestamp": "2019-01-06T22:47:58Z", "text": "This bug is also affecting me on a fresh install of Fedora 29 and Atom 1.33.1.\r\n\r\nIf I try to delete a file within atom, I get the following error:\r\n![gvfs-gio-error](https://user-images.githubusercontent.com/5628330/50742193-8be3d980-11bc-11e9-9d98-8c307d563839.png)\r\n\r\nIf I use gio on the command line:\r\n```\r\n$ gio trash /home/bauthier/Tools/ansible_local/roles/work_packages/files/test.txt\r\n$\r\n```\r\nno issue and it deletes as expected.\r\n\r\nMy system partitions are broken up like so:\r\n```\r\n$ df -h\r\nFilesystem               Size  Used Avail Use% Mounted on\r\ndevtmpfs                 7.8G     0  7.8G   0% /dev\r\ntmpfs                    7.8G  225M  7.6G   3% /dev/shm\r\ntmpfs                    7.8G  2.0M  7.8G   1% /run\r\ntmpfs                    7.8G     0  7.8G   0% /sys/fs/cgroup\r\n/dev/mapper/fedora-root   49G   11G   36G  24% /\r\ntmpfs                    7.8G  236K  7.8G   1% /tmp\r\n/dev/sda2                976M  160M  750M  18% /boot\r\n/dev/mapper/fedora-home  414G  892M  392G   1% /home\r\n/dev/sda1               1022M   21M 1002M   3% /boot/efi\r\ntmpfs                    1.6G  8.4M  1.6G   1% /run/user/1000\r\n```\r\nIf I explicitly set an environment variable like this within my **~/.bashrc**:\r\n```\r\nexport ELECTRON_TRASH=gio\r\n```\r\nit does delete within atom correctly.\r\n\r\nHaving to set this environment variable is less than ideal. I read through cutephoton's research, testing, and analysis and I believe mine works because **/home** and **/** are on different mounts, but the same partition. Perhaps LVM versus standard partitions?\r\n\r\nEither way, I would imagine the handling of \"gvfs\" or \"gio\" would be done by atom and set correctly during start time of the application. I haven't read through the atom code so there may have been a reason to couple that functionality to \"gvfs\" rather than dynamically choosing a suitable trash program and application startup.\r\n", "meta": {"posReactions": "5", "negReactions": "0"}}
{"id": "COM26", "speaker": "yodatak", "conversation_id": "ROOT", "reply_to": "COM25", "timestamp": "2019-01-06T23:11:32Z", "text": "https://github.com/electron/electron/issues/15011\r\n", "meta": {"posReactions": "1", "negReactions": "0"}}
{"id": "COM27", "speaker": "den-is", "conversation_id": "ROOT", "reply_to": "COM26", "timestamp": "2019-01-11T08:44:24Z", "text": "same here with fedora 29 and atom 1.34.0\r\n\r\nthat workaround has helped me https://github.com/atom/atom/issues/15949#issuecomment-434231263", "meta": {"posReactions": "5", "negReactions": "0"}}
{"id": "COM28", "speaker": "abudawud", "conversation_id": "ROOT", "reply_to": "COM27", "timestamp": "2019-01-14T12:58:36Z", "text": "same here with fedora 29 and atom 1.33.0x64\r\n`~$ lsb_release -a`\r\n```\r\nLSB Version:\t:core-4.1-amd64:core-4.1-noarch\r\nDistributor ID:\tFedora\r\nDescription:\tFedora release 29 (Twenty Nine)\r\nRelease:\t29\r\nCodename:\tTwentyNine\r\n```", "meta": {"posReactions": "0", "negReactions": "0"}}
{"id": "COM29", "speaker": "nelson6e65", "conversation_id": "ROOT", "reply_to": "COM28", "timestamp": "2019-01-27T22:23:37Z", "text": "> @histamineblkr If I explicitly set an environment variable like this within my **~/.bashrc**:\r\n> \r\n> ```\r\n> export ELECTRON_TRASH=gio\r\n> ```\r\n\r\nThis did work for me, in Fedora 29. ", "meta": {"posReactions": "1", "negReactions": "0"}}
